Sources/PriorityQueueModule/Heap.swift

```diff
@@ -0,0 +1,537 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+/// A [Min-Max Heap](https://en.wikipedia.org/wiki/Min-max_heap) data structure.
```

Comment:
This is useful information to have in an "Implementation Details" section, but it doesn't tell the reader what a heap is and why they would want to use it. We should add an introduction that is useful for a wider audience.

Something like this would be a good start:

> A collection of comparable items arranged in a way that allows efficient access to its minimal and maximal elements.
>
> Heaps can be used to efficiently provide sorted access to items that arrive incrementally and out of order. For example, a server application may use a heap to store its queue of incoming client transactions, processing them in order of decreasing priority.
>
> Heaps provide highly optimized implementations for the following operations:
>
> - `insert(_:)`: Inserts a new element in O(log(`count`)) time.
> - `min()`/`max()`: Retrieve (but does not remove) the minimum/maximum item in O(1) time.
> - `removeMin()`/`removeMax()`: Remove and return the minimum/maximum item in O(log(`count`)) time.
>
> \<Some toy sample code illustrating these operations\>

Reply:
https://github.com/apple/swift-collections/issues/74